colcon_bundle/verb/__init__.py

```diff
@@ -82,7 +86,29 @@ def check_and_mark_bundle_version(bundle_base, *, this_bundle_version):
             'Bundle version 2 has multiple improvements to bundling speed'
             'and other optimizations, it is highly '
             'recommended.'.format_map(locals()))
-    else:
-        os.makedirs(bundle_base, exist_ok=True)
-
     marker_path.write_text(str(this_bundle_version) + '\n')
+
+
+def check_and_mark_bundle_cache_version(bundle_base: str,
+                                        version: int,
+                                        previously_bundled: bool) -> int:
+    """
+    Check and return the bundle cache version.
+
+    The marker filename is `.bundle_cache_version`.
+
+    :param str bundle_base: The bundle directory
+    :param int version: Version to mark if a version file does not exist
+    :param bool previously_bundled: true if the user has previously used
+    this workspace to build a bundle
+    :returns: the cache layout version to use
+    """
+    marker_path = Path(bundle_base) / '.bundle_cache_version'
```

Comment:
Shall we add marker_path to pathcontext.py as I could see it's referenced multiple places?

Reply:
Marker path has a different value for each marker we place, we are also setting the markers inside the __init__ of PathContext, so we don't have the object yet. I think the __init__ of PathContext is the right place because it is in charge of all the paths and validation of paths.